tensorboard/data/proto/data_provider.proto

```diff
@@ -0,0 +1,295 @@
+import "tensorboard/compat/proto/tensor.proto";
+
+option go_package = "github.com/tensorflow/tensorboard/tensorboard/data/proto/data_provider_proto";
+
+service TensorBoardDataProvider {
+  // List runs within an experiment.
+  rpc ListRuns(ListRunsRequest) returns (ListRunsResponse) {}
+  // List metadata about scalar time series.
+  rpc ListScalars(ListScalarsRequest) returns (ListScalarsResponse) {}
+  // Read data from scalar time series.
+  rpc ReadScalars(ReadScalarsRequest) returns (ReadScalarsResponse) {}
+  // List metadata about tensor time series.
+  rpc ListTensors(ListTensorsRequest) returns (ListTensorsResponse) {}
+  // Read data from tensor time series.
+  rpc ReadTensors(ReadTensorsRequest) returns (ReadTensorsResponse) {}
+  // List metadata about blob sequence time series.
+  rpc ListBlobSequences(ListBlobSequencesRequest)
+      returns (ListBlobSequencesResponse) {}
+  // Read blob references from blob sequence time series. See `ReadBlob` to read
+  // the actual blob data.
+  rpc ReadBlobSequences(ReadBlobSequencesRequest)
+      returns (ReadBlobSequencesResponse) {}
+  // Read data for a specific blob.
+  rpc ReadBlob(ReadBlobRequest) returns (stream ReadBlobResponse) {}
+}
+
+message ListRunsRequest {
+  // ID of experiment in which to query data.
+  string experiment_id = 1;
+}
+
+message ListRunsResponse {
+  repeated Run runs = 1;
+}
+
+message Run {
+  // Unique opaque string identifier.
+  string id = 1;
+  // User-facing name.
+  string name = 2;
+  // Wall time of earliest recorded event, as floating-point seconds since
+  // epoch (same as event file format).
+  double start_time = 3;
+}
+
+message ListScalarsRequest {
+  // ID of experiment in which to query data.
+  string experiment_id = 1;
+  // Required filter for plugin name. If omitted, an empty message is implied.
+  PluginFilter plugin_filter = 2;
+  // Optional filter for time series. If omitted, all time series match.
+  RunTagFilter run_tag_filter = 3;
+}
+
+message ListScalarsResponse {
+  repeated RunEntry runs = 1;
+  message RunEntry {
+    string run_name = 1;
+    repeated TagEntry tags = 2;
+  }
+  message TagEntry {
+    string tag_name = 1;
+    ScalarMetadata time_series = 2;
+  }
+}
+
+message PluginFilter {
+  // Only match data with exactly this plugin name.
+  string plugin_name = 1;
+}
+
+message RunTagFilter {
+  // Optional filter for runs. If omitted, all runs match.
+  RunFilter runs = 1;
+  // Optional filter for tags. If omitted, all tags match.
+  TagFilter tags = 2;
+}
+
+message RunFilter {
+  // Only match runs with exactly one of these names. In particular, if this
+  // list is empty, no runs match.
+  repeated string runs = 1;
+}
+
+message TagFilter {
+  // Only match tags with exactly one of these names. In particular, if this
+  // list is empty, no tags match.
+  repeated string tags = 1;
+}
+
+message ScalarMetadata {
+  // Largest step value of any datum in this time series. Non-negative.
```

Comment:
Will we actually enforce that this is non-negative?  AFAICT we use int64 everywhere so it certainly can be negative on the wire; if a time series only has negative steps, does that mean we return 0 here?